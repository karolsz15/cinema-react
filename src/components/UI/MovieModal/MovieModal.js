import React from 'react';
import PropTypes from 'prop-types';
import Modal from "react-bootstrap/Modal";
import Button from 'react-bootstrap/Button';

const MovieModal = props => {

  const {title, trailer, onHide} = props;

  return (
    <Modal
      {...props}
      size="lg"
      aria-labelledby="contained-modal-title-vcenter"
      centered
    >
      <Modal.Header closeButton>
        <Modal.Title id="contained-modal-title-vcenter">
          {title}
        </Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <iframe 
          src={trailer} 
          title={title} 
          width="560" 
          height="315" 
          frameBorder="0" 
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowFullScreen>
        </iframe>
      </Modal.Body>
      <Modal.Footer>
        <Button onClick={onHide}>Close</Button>
      </Modal.Footer>
    </Modal>
  );
};

MovieModal.propTypes = {
  title: PropTypes.string,
  trailer: PropTypes.string,
  onHide: PropTypes.func.isRequired
};

export default MovieModal;