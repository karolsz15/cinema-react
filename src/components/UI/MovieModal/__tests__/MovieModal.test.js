import React from 'react';
import { shallow } from 'enzyme';
import MovieModal from '../MovieModal';

let wrapper = shallow(<MovieModal />);

describe('MovieModal component', () => {
  it('should render the MovieModal correctly', () => {   
    expect(wrapper).toMatchSnapshot();
  });
});