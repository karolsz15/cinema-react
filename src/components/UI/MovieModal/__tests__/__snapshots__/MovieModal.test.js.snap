// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieModal component should render the MovieModal correctly 1`] = `
<Bootstrap(Modal)
  aria-labelledby="contained-modal-title-vcenter"
  centered={true}
  size="lg"
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      id="contained-modal-title-vcenter"
    />
  </ModalHeader>
  <ModalBody>
    <iframe
      allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
      allowFullScreen={true}
      frameBorder="0"
      height="315"
      width="560"
    />
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      type="button"
      variant="primary"
    >
      Close
    </Button>
  </ModalFooter>
</Bootstrap(Modal)>
`;
